File: src/main/java/com/ragicriSushi/pw/DAO/PiattoOrdinato.java
package com.ragicriSushi.pw.DAO;

import lombok.*;

import javax.persistence.*;

import static javax.persistence.GenerationType.IDENTITY;

@Entity
@Table(name = "PiattoOrdinato")
@NoArgsConstructor
@Getter
@Setter
@Builder
@AllArgsConstructor
public class PiattoOrdinato {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    int id;

    @ManyToOne
    @JoinColumn
    OrdinazioneDAO ordinazione;

    @ManyToOne
    @JoinColumn
    PiattoDAO piatto;

    @Column
    boolean consegnato = false;

    @Column
    int quantita;

    public boolean getConsegnato(){
        return this.consegnato;
    }

}

File: src/app/Theme/piattoOrdinato/PiattoOrdinato.ts
import {Piatto} from '../piatto/Piatto';
import {Ordinazione} from '../Ordinazione/Ordinazione';

export class PiattoOrdinato {

    id: number;
    ordinazione: Ordinazione;
    piatto: Piatto;
    consegnato: boolean;
    quantita: number;
}

